### PR TITLE
Attempt to generate "docs" during CircleCI tests.

### PR DESCRIPTION
**Pull request by abernix**
There have been a number of occasions (most recently, Friday; see https://github.com/meteor/docs/commit/8f7ceb5de5bf9bdd5935e3447164ce8f24cece7b and https://github.com/meteor/docs/commit/fc5396a52f6fb7c8963f1b280e1ded1c8722134d) where I have gone to bump the Meteor docs to the next major version, only to find that the docs deployment (which occurs in a separate repository; found at https://github.com/meteor/docs) fails due to code-refactoring which has taken place in its submodule (`code`, which is this Meteor codebase).

In order to provide early warning when a JSDoc declaration in the Meteor code is going to (eventually) break the docs deployment, this commit introduces a simple checkout and doc generation, in the same manner as the docs would in its own repository, within the CircleCI workflow.

It's been configured in a way where it will not stop the normal Meteor test-suite from running, however it will show a separate indicator on GitHub pull-requests as to whether the generation of docs was successful or not.

---

**Edit:** As expected, the (new) **Docs** step is [failing](https://circleci.com/gh/meteor/meteor/15085) on this PR:

```
  Error: Cannot render apibox without API data: Meteor.status
```

### CODE REVIEW

**Comment by abernix** [comment]
The combination of the two follow-up commits I've made on this PR (https://github.com/meteor/meteor/pull/9593/commits/ababb18aef6d75055f1f460ae4709c2ef26c4836 and https://github.com/meteor/meteor/pull/9593/commits/4d861735c5c0ad43a21de0bc7d52c1a07be1ba62) fix the problem with docs generation, and to further prove the success of the test functionality, the tests on this PR are green with the addition of the latter commit.  I hope this will prevent future JSDoc regressions, which are usually unfortunate blindsides to otherwise great code improvements (e.g. ECMAScript++, structure, etc.).

This PR should be ready for final review now.

File: packages/ddp-client/common/livedata_connection.js [hunk 1]
```diff
@@ -330,6 +330,7 @@ export class Connection {
   /**
    * @memberOf Meteor
    * @importFromPackage meteor
+   * @alias Meteor.subscribe
```
Comment (abernix) [inline packages/ddp-client/common/livedata_connection.js]: For the sake of posterity...

The alternative to `@alias` I had investigated was the use of [`@borrows`](http://usejsdoc.org/tags-borrows.html) declarations in `ddp-client/client/client_convenience.js`, where the `Meteor.{subscribe,methods,call,apply,status,reconnect,disconnect}` methods are aliased to the connection:

https://github.com/meteor/meteor/blob/9daf412b2b7f69c068319ba3f9762817f210eea1/packages/ddp-client/client/client_convenience.js#L52-L64

In addition to each of those `@borrows` needing a `@namespace Meteor` declaration, they would have had to be partially duplicated in `packages/ddp-server/server_convenience.js`, which would have resulted in duplicate declarations in the resulting `jsdoc` `data/data.json` file, most likely confusing our rather primitive `apibox`es on docs:

https://github.com/meteor/meteor/blob/9daf412b2b7f69c068319ba3f9762817f210eea1/packages/ddp-server/server_convenience.js#L12-L17

Yeah, so I chose the `@alias` method.

File: packages/ddp-client/common/livedata_connection.js [hunk 2]
```diff
@@ -888,10 +891,11 @@ export class Connection {
   }
 
   /**
-   * @summary Get the current connection status. A reactive data source.
-   * @locus Client
    * @memberOf Meteor
    * @importFromPackage meteor
+   * @alias Meteor.status
+   * @summary Get the current connection status. A reactive data source.
```
Comment (abernix) [inline packages/ddp-client/common/livedata_connection.js]: Reordered `@` declarations for consistency only.

**Comment by abernix** [comment]
In proof that I wasn't immediately expecting of how important this PR is, I merged `devel` into this to solve unrelated test failures and this PR's (new) "Docs" test started failing because of another missing JSDoc declaration from another commit.  (Fixed in this PR with https://github.com/meteor/meteor/pull/9593/commits/46eaab39b6211a5ed56d544461cab1959ccb3c85) Wheeee! 🎢 

When this is passing, I'll merge this to avoid additional problems from cropping up with the plethora of [modernization PRs](https://github.com/meteor/meteor/pulls?utf8=%E2%9C%93&q=is%3Apr+sort%3Aupdated-desc+in%3Atitle+modernize+created%3A2018-01-20..2018-02-01) (which oft change `function` constructors to native ECMAScript `class`es) we've received from our wonderful contributors. ✋👉🙏👈✋

**Comment by abernix** [comment]
The last several runs (and re-runs) of this PR have consistently failed various `tinytest - observeChanges` tests: https://travis-ci.org/meteor/meteor/builds/336029193#L2750. It shouldn't be related to these changes, but needs to be investigated.

**Comment by jamesmillerburgess** [comment]
@abernix Great that you've automated some of this tedious work! I'll rebase all of my PRs once this gets merged and correct any docs failures.

**Pull request merged**